metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -362,63 +438,77 @@ final class Doctor(
-      buildTargets.scalaTarget(targetId).map(extractScalaTargetInfo).toList
-    val javaDoctorInfo =
-      buildTargets.javaTarget(targetId).map(extractJavaTargetInfo).toList
-    scalaDoctorInfo ::: javaDoctorInfo
+    val javaTarget = buildTargets.javaTarget(targetId)
+    buildTargets
+      .scalaTarget(targetId)
+      .map(st => extractScalaTargetInfo(st, javaTarget))
+      .toList
   }
 
-  private def extractScalaTargetInfo(target: ScalaTarget) = {
+  private def extractScalaTargetInfo(
+      target: ScalaTarget,
+      javaTarget: Option[JavaTarget]
+  ) = {
     val scalaVersion = target.scalaVersion
-    val definition: String =
+    val interactive =
       if (mtagsResolver.isSupportedScalaVersion(scalaVersion)) {
-        Icons.unicode.check
+        DoctorStatus(Icons.unicode.check, isCorrect = true)
       } else {
-        Icons.unicode.alert
+        DoctorStatus(Icons.unicode.error, isCorrect = false)
       }
-    val completions: String = definition
     val isSemanticdbNeeded = !target.isSemanticdbEnabled
-    val references: String =
+    val indexes =
       if (isSemanticdbNeeded) {
-        Icons.unicode.alert
+        DoctorStatus(Icons.unicode.error, isCorrect = false)
       } else {
-        Icons.unicode.check
+        DoctorStatus(Icons.unicode.check, isCorrect = true)
       }
     val recommendedFix = problemResolver.recommendation(target)
-    DoctorTargetInfo(
-      target.displayName,
-      target.dataKind,
-      target.baseDirectory,
-      scalaVersion,
-      definition,
-      completions,
-      references,
-      recommendedFix
-    )
-  }
-
-  private def extractJavaTargetInfo(target: JavaTarget) = {
-    val scalaVersion = "Java"
-    val definition: String = Icons.unicode.check
-    val completions: String = Icons.unicode.alert
-    val isSemanticdbNeeded = !target.isSemanticdbEnabled
-    val references: String =
-      if (isSemanticdbNeeded) {
-        Icons.unicode.alert
-      } else {
-        Icons.unicode.check
+    val (targetType, diagnostics) =
+      target.sbtVersion match {
+        case Some(sbt) =>
+          (s"sbt $sbt", DoctorStatus(Icons.unicode.alert, isCorrect = false))
+        case None =>
+          (
+            s"Scala $scalaVersion",
+            DoctorStatus(Icons.unicode.check, isCorrect = true)
+          )
       }
-    val recommendedFix = problemResolver.recommendation(target)
+    val (javaSupport, javaRecommendation) = javaTarget match {
+      case Some(target) if target.isSemanticdbEnabled =>
+        (DoctorStatus(Icons.unicode.check, isCorrect = true), None)
+      case Some(target) =>
+        (
+          DoctorStatus(Icons.unicode.alert, isCorrect = false),
+          problemResolver.recommendation(target)
+        )
+      case None => (DoctorStatus(Icons.unicode.alert, isCorrect = false), None)
+    }
+
+    val canRun = target.info.getCapabilities().getCanRun()
+    val canTest = target.info.getCapabilities().getCanTest()
```

Comment:
@kpodsiad I remember that there were issue with these capabilities values in bloop and sbt. Are they reliable now?

Reply:
run and test are reliable, only debug was problematic